File: Problems/Strings/Valid-Anagram/Valid-Anagram.cs
using System;

// Using an array - Time: O(n)
public class Solution
{
    public bool IsAnagram(string s, string t)
    {
        int[] cnt = new int[26];

        foreach (char c in s)
            cnt[c - 'a']++;

        foreach (char c in t)
            cnt[c - 'a']--;

        foreach (int n in cnt)
        {
            if (n != 0)
            {
                return false;
            }
        }

        return true;
    }
}

class Program
{
    static void Main(string[] args)
    {
        string s = "anagram";
        string t = "nagaram";
        Console.WriteLine("Input: s = " + s + ", t = " + t);

        Solution sol = new Solution();
        bool result = sol.IsAnagram(s, t);

        Console.WriteLine("Output: " + result);
    }
}

File: Problems/Strings/Valid-Anagram/Valid-Anagram.cpp
#include <iostream>
#include <string>

// Using an array - Time: O(n)

class Solution {
public:
    bool isAnagram(std::string s, std::string t)
    {
        int cnt[26] = {};

        for (char c : s)
            cnt[c - 'a']++;

        for (char c : t)
            cnt[c - 'a']--;

        for (int n : cnt)
        {
            if (n)
                return false;
        }

        return true;
    }
};

int main() 
{
    std::string s = "anagram";
    std::string t = "nagaram";
    std::cout << "Input: s = " << s << ", t = " << t << std::endl;

    Solution sol;
    bool result = sol.isAnagram(s, t);

    std::cout << "Output = " << std::boolalpha << result << std::endl;

    return 0;
}
